### PR TITLE
Use EditTextInfo for RichEdit50 controls

### LINKED ISSUE #13420: NVDA can't make system cursor and review cursor follow by using say all in WX's rich text box, Take NVDA log viewer and bookworm as examples
### Steps to reproduce:
This issue comes from [this discussion][2].
1. [Download][1] Bookworm2022.1a5 portable and open a document at will, or Or open the NVDA log viewer;
2. Press NVDA+Down Arrow in the text of document;
3. Press the Numpad Plus in the text of document;
4. Press twice NVDA+shift+numpad minus;

### Actual behavior:
Perform step 2, the system cursor does not change;
Perform step 3, the review cursor does not change;
Perform step 4, Unable to set the system cursor to the position of the review cursor.

### Expected behavior:
Perform step 2 and the system cursor should follow the movement
Perform step 3 and the review cursor should follow the movement
Perform step 4 the system cursor can be set to the position of the review cursor

#### NVDA installed/portable/running from source:
installed
#### NVDA version:
2021.3.3
#### Windows version:
Windows 10 21H2 (x64) build 19044.1526
#### Name and version of other software in use when reproducing the issue:
Bookworm2022.1 A5 and NVDA log viewer
#### Other information about your system:
None
#### Does the issue still occur after restarting your computer?
yes
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
The latest alpha version of NVDA can also reproduce
#### If NVDA add-ons are disabled, is your problem still occurring?
Yes
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
Yes

[1]: https://github.com/blindpandas/bookworm/releases
[2]: https://github.com/blindpandas/bookworm/discussions/116

### LINKED ISSUE #13927: Say all does not turn the page properly in bookworm
### Steps to reproduce:
1. Download Bookworm, a book reader. A build can be found here. https://ci.appveyor.com/api/buildjobs/kffqbo4qru8pok39/artifacts/scripts%2FBookworm-2022.1b1-x64-portable.zip. Go into general preferences by pressing control+shift+p, and make sure the Try to support the screen reader's continuous reading mode by automatically turning pages (may not work in some cases) option is enabled.
2. Open a PDF file and read it using say all. Example: https://www.dropbox.com/s/rlq00oplvkfzfl2/DoubleTalk%20RC8660%20Chipset%20Manual.pdf?dl=1.
### Actual behavior:
Say all doesn't turn the page after the end of it is reached. You will notice that it will stop reading around the end of page 1.
### Expected behavior:
Say all should be able to continuously turn the page.
#### NVDA installed/portable/running from source:
Installed
#### NVDA version:
alpha-25928,3894e25f
#### Windows version:
Windows 11, Version 22H2 (OS Build 25158.1000).
#### Name and version of other software in use when reproducing the issue:
Bookworm 2022.1b1.
#### Other information about your system:
N/A
#### Does the issue still occur after restarting your computer?
Yes.
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
Yes, NVDA snapshot alpha-25824,9172c09a and below work normally.
#### If NVDA add-ons are disabled, is your problem still occurring?
Yes.
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
Yes.

### PR DESCRIPTION
**Pull request by seanbudd**
Opening draft PR to start build

### Link to issue number:
Fixes #13420, #13927

### Summary of the issue:
TODO

### Description of user facing changes
Review cursor and system cursor tracking fixed for RichEdit50 controls (e.g. NVDA log viewer, Bookworm).

### Description of development approach
Use EditTextInfo for RichEdit50 controls.

### Testing strategy:
TODO

### Known issues with pull request:
Bookworm sayAll stops after each page. This is different behaviour to 2022.2, where sayAll continues on the new page.
This is better than the current behaviour on master, which causes sayAll to be caught in an infinite loop scrolling at the bottom of the last page.

### Change log entries:
Bug fixes

### Code Review Checklist:

- [ ] Pull Request description:
  - description is up to date
  - change log entries
- [ ] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [ ] API is compatible with existing add-ons.
- [ ] Documentation:
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [ ] UX of all users considered:
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English

### CODE REVIEW

**Comment by leonardder** [comment]
Please note that Wordpad also has a RichEdit50 and there, it works just fine. That made me think that it is just the ITextDocument implementation in WX that's broken. On the contrary, the UIA implementation works properly, I wonder what that uses under the hood.
Anyhow, I think this pr should be limited to WX windows only, i.e. there should be a wx window class on one of the parent windows.

**Comment by lukaszgo1** [comment]
I'm not sure this PR is a right approach:
- While #13420 is problematic the main issue we should be trying to solve is #13927 and that almost certainly don't require such drastic change as this - it regressed in #13670 and the issue is somewhere in `sayAllHandler`.
- As mentioned by Leonard we would break all non WX applications using RichEdit50 i.e. Wordpad.
- Even for WX rich edit controls this causes regressions - links cannot be detected and automatic language detection is broken.


In short it seems more reasonable to fix the regression from #13670 for 2022.3, and then try to identify what goes wrong when setting caret in WX RichEdit - perhaps it would be possible to use method for setting caret from non richedit controls and keep the rest of the support from richedit.
